src/ol/style/fill.js

```diff
@@ -31,6 +31,18 @@ ol.style.Fill = function(opt_options) {
 
 
 /**
+ * Clones the style.
+ * @return {ol.style.Fill} The cloned style.
+ * @api
+ */
+ol.style.Fill.prototype.clone = function() {
+  return new ol.style.Fill({
+    color: (this.getColor() instanceof Array) ? this.getColor().slice(0) : this.getColor()
```

Comment:
I think checking for `slice` is fine (and quite a bit nicer than `if (typeof color === 'string' || Array.isArray(color) || color instanceof Uint8Array || color instanceof Uint8ClampedArray)`).

As a side note, it looks like `ol.color.asArray()` wouldn't work when called with a `Uint8Array` or `Uint8ClampedArray`.  I'll create a ticket for that.